docs/examples/nmt/nmt/dataset.py

```diff
@@ -0,0 +1,70 @@
+# coding: utf-8
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint:disable=redefined-outer-name,logging-format-interpolation
+"""Translation datasets."""
+
+
+__all__ = ['TOY']
+
+import os
+from gluonnlp.data.translation import _TranslationDataset, _get_pair_key, _get_home_dir
+from gluonnlp.data.registry import register
+
+
+@register(segment=['train', 'val', 'test'])
+class TOY(_TranslationDataset):
```

Comment:
Don't use TOY dataset. This one is only created for testing the script. It is not calibrated with true training/testing sets.

Reply:
The performance in TOY dataset is not meaningful. This one only makes sure that the script is runnable. Instead, to present the meaningful result using the pre-trained model, you can consider  testing the first 100 sentences in testing set by setting a stop criterion in evaluation function.

Reply:
Thanks for noticing this. But I have considered the case when creating this notebook, the notebook is just designed for showcase. The inference on WMT2014 will take more than 5mins, it is not possible to show to the audience in the tutorial. However, we have the dataset confirguration in the hyperparameter.py, we can tell the audience they can change the dataset from TOY to WMT2014 to reproduce the results. The notebook decription should have already reflected that. I have already synced this with Leo @leezu .

Reply:
@cgraywang but according to @szhengac it seems that the Toy dataset is not meaningful, in fact it is not a "well known" dataset but just some test data that @szhengac extracted. It may not even follow the same preprocessing as WMT14. How about using the results on the first X sentences of WMT14 instead so that people don't get confused about the dataset part?

Reply:
@szhengac @leezu  I just had a new commit which involves our discussion by using the WMT test dataset to show the results of the pretrained transformer. And use the TOY dataset just to showcase how to train the Transformer. Plus I also added the translation inference based on the user provided example and the Bleu score evaluation. Thanks.

Reply:
To allow the inference on user example, the model requires specific preprocessing pipeline on the sentence, including mosesdecoder processing shown in machine_translation.rst and applying BPE. As the learned BPE file is not stored, we need to preprocess the plain training set one more time and get the BPE file.

Reply:
@szhengac I have already written a simple inference function to do the data transformation and use beam search to generate the target sentence. Please check in the notebook example.